lib/protoflow-blocks/src/blocks/text/concat_strings.rs

```diff
@@ -0,0 +1,130 @@
+// This is free and unencumbered software released into the public domain.
+
+extern crate std;
+
+use crate::{
+    prelude::{String, Vec, vec},
+    StdioConfig, StdioError, StdioSystem, System,
+};
+use protoflow_core::{
+    Block, BlockResult, BlockRuntime, InputPort, OutputPort,
+};
+use protoflow_derive::Block;
+use simple_mermaid::mermaid;
+
+/// A block that concat strings.
+///
+/// # Block Diagram
+#[doc = mermaid!("../../../doc/text/concat_strings.mmd")]
+///
+/// # Sequence Diagram
+#[doc = mermaid!("../../../doc/text/concat_strings.seq.mmd" framed)]
+///
+/// # Examples
+///
+/// ## Using the block in a system
+///
+/// ```rust
+/// # use protoflow_blocks::*;
+/// # fn main() {
+/// System::build(|s| {
+///     let joiner = ",";
+///     let stdin = s.read_stdin();
+///     let line_decoder = s.decode_with(config.encoding);
+///     let concat_strings = s.concat_strings_by(&joiner);
+///     let line_encoder = s.encode_with(config.encoding);
+///     let stdout = config.write_stdout(s);
+///     s.connect(&stdin.output, &line_decoder.input);
+///     s.connect(&line_decoder.output, &concat_strings.input);
+///     s.connect(&concat_strings.output, &line_encoder.input);
+///     s.connect(&line_encoder.output, &stdout.input);
+/// });
+/// # }
+/// ```
+///
+/// ## Running the block via the CLI
+///
+/// ```console
+/// $ protoflow execute ConcatStrings joiner=","
+/// ```
+///
+#[derive(Block, Clone)]
+pub struct ConcatStrings {
+    /// The input message string stream.
+    #[input]
+    pub input: InputPort<String>,
+    /// The output string.
+    #[output]
+    pub output: OutputPort<String>,
+    /// A parameter placed between each input parameter
+    #[parameter]
+    pub joiner: String
```

Comment:
Use "delimiter" everywhere instead of "joiner"

Reply:
done